src/client/dfs/dfs.c

```diff
@@ -2184,9 +2185,12 @@ dfs_mkdir(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 
 	rc = insert_entry(parent->oh, th, name, len,
 			  DAOS_COND_DKEY_INSERT, &entry);
-	if (rc != 0) {
+	if (rc == EEXIST) {
 		daos_obj_close(new_dir.oh, NULL);
 		return rc;
+	} else if (rc != 0) {
+		daos_obj_close(new_dir.oh, NULL);
+		return daos_der2errno(rc);
```

Comment:
im not sure i understand this change. rc is already errno so why convert again if rc != 0 ?